common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/SimpleDownloadFile.java

```diff
@@ -46,8 +46,9 @@ public SimpleDownloadFile(File file, TransportConf transportConf) {
   }
 
   @Override
-  public boolean delete() {
-    return file.delete();
```

Comment:
Do you mean the original implementation has the reference to `DownloadFile` instance itself? `DownloadFile` -> `file`?

Reply:
Yep.

Reply:
Not the `SimpleDownloadFile`, but the `ReferenceWithCleanup` to `file`.

Reply:
```
def cleanUp(): Unit = {
        logDebug(s"Clean up file $filePath")

        if (!file.delete()) {                                                       <--- here
          logDebug(s"Fail to delete file $filePath")
        }
}
```